src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -1083,10 +1121,11 @@ private boolean tryWindBack(final int cycle) {
     }
 
     void index0(final long index) {
-        if (indexValue == null)
+        if (indexValue == null) {
             this.index = index;
-        else
-            indexValue.setValue(index);
+        } else if (!indexValue.compareAndSwapValue(this.indexChecker, index)) {
+            this.indexChecker = Long.MIN_VALUE; // invalid.
```

Comment:
Would it be better to give this a constant with a meaningful name? It's a bit hard to follow what the significance of Long.MIN_VALUE is.